l10n_it_asset_management/models/asset_depreciation.py

```diff
@@ -494,15 +483,46 @@ def get_depreciation_amount_multiplier(
                     start_date=dt_start_str,
                 )
             )
+        fiscal_year_obj = self.env["account.fiscal.year"]
+        fy_start = fiscal_year_obj.get_fiscal_year_by_date(
+            date_start, company=self.company_id
+        )
+        fy_dep = fiscal_year_obj.get_fiscal_year_by_date(
+            dep_date, company=self.company_id
+        )
 
-        if self.pro_rata_temporis or self._context.get("force_prorata"):
-            fiscal_year_obj = self.env["account.fiscal.year"]
-            fy_start = fiscal_year_obj.get_fiscal_year_by_date(
-                date_start, company=self.company_id
-            )
-            fy_dep = fiscal_year_obj.get_fiscal_year_by_date(
-                dep_date, company=self.company_id
+        # Base multiplier
+        multiplier = self.percentage / 100
+
+        if period == "month":
+            multiplier /= 12
+
+        if period_count:
+            multiplier *= period_count
+
+        # Set multiplier from pro-rata temporis with date_end
+        if (
+            self.pro_rata_temporis or self._context.get("force_prorata")
+        ) and self.date_end:
+            fy_end = fiscal_year_obj.get_fiscal_year_by_date(
+                self.date_end, company=self.company_id
             )
+            total_days = (self.date_end - self.date_start).days + 1
+            dep_year_days = (fy_dep.date_to - fy_dep.date_from).days + 1
```

Comment:
Questi calcoli vengono già fatti in `get_pro_rata_temporis_multiplier` e `get_pro_rata_temporis_dates`, si possono modificare quei metodi invece di aggiungere qui nuovo codice? In tutti i metodi è sempre disponibile `self`, quindi dovrebbe essere possibile modificarli in base al nuovo `self.date_end`.
Così tutti i calcoli per il caso pro-rata rimangono in quei metodi, che ne pensi?